conans/client/toolchain/msbuild.py

```diff
@@ -2,10 +2,44 @@
 import textwrap
 from xml.dom import minidom
 
+from conans.client.toolchain.visual import vcvars_arch, vcvars_command
+from conans.client.tools import msvs_toolset
 from conans.errors import ConanException
 from conans.util.files import save, load
 
 
+class MSBuildCmd(object):
+    def __init__(self, conanfile, sln):
+        self._sln = sln
+        self._conanfile = conanfile
+        self.version = conanfile.settings.get_safe("compiler.version")
+        self.vcvars_arch = vcvars_arch(conanfile)
+        self.build_type = conanfile.settings.get_safe("build_type")
+        msvc_arch = {'x86': 'x86',
+                     'x86_64': 'x64',
+                     'armv7': 'ARM',
+                     'armv8': 'ARM64'}
+        # if platforms:
+        #    msvc_arch.update(platforms)
+        arch = conanfile.settings.get_safe("arch")
+        msvc_arch = msvc_arch.get(str(arch))
+        if conanfile.settings.get_safe("os") == "WindowsCE":
+            msvc_arch = conanfile.settings.get_safe("os.platform")
+        self.platform = msvc_arch
+ 
+    def command(self):
+        vcvars = vcvars_command(self.version, architecture=self.vcvars_arch,
```

Comment:
Just to confirm: Is vcvars called automatically when using the visual studio IDE?

Reply:
Yes, when you double click on a VS .sln and the IDE opens, it will manage its own environment and vcvars. It is just from the command line that you need it.